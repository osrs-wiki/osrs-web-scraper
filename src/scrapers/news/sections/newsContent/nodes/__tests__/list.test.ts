import { MediaWikiBuilder } from "@osrs-wiki/mediawiki-builder";
import parse from "node-html-parser";

import listParser from "../list";

describe("list node", () => {
  test("list should parse and render", () => {
    const root = parse("<ul><li>test1</li><li>test2</li></ul>");
    const builder = new MediaWikiBuilder();
    builder.addContents([listParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("list should ignore breaks", () => {
    const root = parse("<ul><li>test1</li><br/><li>test2</li><br/></ul>");
    const builder = new MediaWikiBuilder();
    builder.addContents([listParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });
});
