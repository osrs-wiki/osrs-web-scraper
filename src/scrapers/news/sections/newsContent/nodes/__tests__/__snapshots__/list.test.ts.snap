// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list node list should ignore breaks 1`] = `
"
* test1
* test2

"
`;

exports[`list node list should parse and render 1`] = `
"
* test1
* test2

"
`;
